### ISSUE TITLE
Feature request: Add resource for managing Service Bus Geo-Replication Pairing

### ISSUE THREAD

**Issue by shopskyi** [terraform-providers/terraform-provider-azurerm#3136]
### Community Note

* Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As a DevOps, I want to be able to manage Service Bus Geo-Replication Pairings using terraform so that I don't have to automate it using PowerShell/Azure CLI.  Service Bus Geo-Replication Pairings provide Geo-disaster recovery for Service Bus.

### New or Affected Resource(s)

* azurerm_servicebus_namespace_pairing
* DATA:azurerm_servicebus_namespace_pairing

### Potential Terraform Configuration

```
resource "azurerm_servicebus_namespace_pairing" "pairing" {
  name = "service-bus-alias"
  resource_group = "<resource_group_name>"
  location = "<region>"

  primary_service_bus_namespace = "service-bus-primary"
  secondary_service_bus_namespace = "service-bus-secondary"
}

data "azurerm_servicebus_namespace_pairing" "pairing" {
  name = "service-bus-alias"
  resource_group = "<resource_group_name>"
}

output "service_bus_alias_primary_key" {
  value = "${data.azurerm_servicebus_namespace_pairing.pairing.primary_key}"
}
output "service_bus_alias_primary_connection_string" {
  value = "${data.azurerm_servicebus_namespace_pairing.pairing.primary_connection_string}"
}
output "service_bus_alias_secondary_key" {
  value = "${data.azurerm_servicebus_namespace_pairing.pairing.secondary_key}"
}
output "service_bus_alias_secondary_connection_string" {
  value = "${data.azurerm_servicebus_namespace_pairing.pairing.secondary_connection_string}"
}
```

### References

- Information about Azure Service Bus Geo-disaster recovery available here -  https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-geo-dr
- APIs available here - https://docs.microsoft.com/en-us/rest/api/servicebus/disasterrecoveryconfigs

**Comment by Tbohunek** [comment]
Any estimates on when this might get implemented?

**Comment by sebATOS49** [comment]
Any ideas to perform this pairing with actual resources?

**Comment by karanratra** [comment]
Is this feature to use terraform to configure Azure Service Bus Pairing available or we have to use the portal still to configure DR for ASB ?

**Comment by shopskyi** [comment]
Any updates?

**Comment by sti291271** [comment]
Is this topic will be available soon?

**Comment by dylan-asos** [comment]
I'm having to do a horrible hack right now to work around this, so have picked this up. Changes currently on my fork at https://github.com/terraform-providers/terraform-provider-azurerm/compare/master...dylan-asos:master  

Still got to work through testing which I'll do this weekend to get it to a stable point, but just flagging for visibility here + would hope to raise a PR over the next week or so.

**Comment by dylan-asos** [comment]
@shopskyi PR open at #11638 - slightly different from your suggested resource name as there was an existing DR config resource for event hubs, and it made sense for consistency to keep them closely aligned

**Comment by dylan-asos** [comment]
Released in version 2.59 of the provider - https://github.com/terraform-providers/terraform-provider-azurerm/pull/11638#issuecomment-840973560
